fetcher/src/job/job.utils.ts

```diff
@@ -234,3 +237,47 @@ export function shouldReport(
     return false
   }
 }
+
+export async function fetchDataFromDex(adapterList, decimals: number, logger) {
+  const data = await Promise.allSettled(
+    adapterList.map(async (adapter) => {
+      try {
+        let datum: number
+        if (adapter.type == 'UniswapPool') datum = await extractKlaySwapPrice(adapter, decimals)
+        checkDataFormat(datum)
+        return { id: adapter.id, value: datum }
+      } catch (e) {
+        logger.error(`Fetching with proxy ${adapter.proxy.host} failed in ${adapter.name}`)
+        logger.error(e)
+        throw e
+      }
+    })
+  )
+  return data.flatMap((D) => (D.status == 'fulfilled' ? [D.value] : []))
+}
+
+export async function extractKlaySwapPrice(adapter, decimals) {
+  const provider = await providerByChain(adapter.chain)
+  const poolContract = new Contract(adapter.address, uniswapPoolAbis, provider)
+  const rawData = await poolContract.slot0()
+  const datum = sqrtPriceX96ToTokenPrice(
+    BigInt(rawData[0]),
+    adapter.token0Decimals,
+    adapter.token1Decimals
+  )
+  return Math.round(datum * 10 ** decimals)
+}
+
+export async function providerByChain(chainId: number) {
+  if (chainId == 8217) {
```

Comment:
Rather than hardcoding chain id '8217' here, it'd be nice if const is declared for chain id